jobs.py

```diff
@@ -124,103 +138,65 @@ def new_job(self, build_url, build_id=None, changeset=None, tree=None,
-            test.generate_guid()
-        test_names.sort()
 
-        try:
-            conn = self._conn()
-            job_cursor = conn.cursor()
-            test_cursor = conn.cursor()
-            job_cursor.execute(
-                'select id from jobs where device=? and build_url=?',
-                (device, build_url))
-            job = job_cursor.fetchone()
-            while job:
-                job_id = job[0]
-                job_test_names = [
-                    test_row[0] for test_row in
-                    test_cursor.execute(
-                        'select name from tests where jobid=?',
-                        (job_id,))]
-                job_test_names.sort()
-                if test_names == job_test_names:
-                    duplicate = True
-                    break
-                job = job_cursor.fetchone()
-        except sqlite3.OperationalError:
-            logger.exception('jobs.new_job')
-            pass
-
-        if duplicate:
-            logger.warning('jobs.new_job: Not adding duplicate job: '
-                           'build: %s, device: %s, test_names: %s' % (
-                               build_url, device, test_names))
-            return
+        conn = self._conn()
+        job_cursor = self._execute_sql(
+            conn,
+            'select id from jobs where device=? and build_url=?',
+            values=(device, build_url))
 
-        if not job_id:
-            attempt = 0
-            while True:
-                attempt += 1
-                try:
-                    conn = self._conn()
-                    job_cursor = conn.cursor()
-                    job_cursor.execute(
-                        'insert into jobs values '
-                        '(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, 0, ?)',
-                        (None, now, None, build_url, build_id, changeset, tree,
-                         revision, revision_hash, enable_unittests, device))
-                    conn.commit()
-                    job_id = job_cursor.lastrowid
-                    break
-                except sqlite3.OperationalError:
-                    email_sent = self.report_sql_error(
-                        attempt, email_sent,
-                        'Unable to insert job into jobs database.',
-                        'Please check the logs for full details.',
-                        'Attempt %d failed to insert job (%s, %s, %s) into '
-                        'database. Waiting for %d seconds.' %
-                        (attempt, build_url, device, test_names,
-                         self.SQL_RETRY_DELAY))
+        job = job_cursor.fetchone()
+        job_cursor.close()
+        if job:
+            job_id = job[0]
+        else:
+            job_cursor = self._execute_sql(
+                conn,
+                'insert into jobs values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, 0, ?)',
+                values=(None, now, None, build_url, build_id, changeset, tree,
```

Comment:
I'm not sure. It wouldn't be an explicit None since SQL won't understand it. I used None as an argument so pysqlite would handle it properly. I'd rather leave it alone for now.